device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceIO.java

```diff
@@ -37,11 +33,13 @@
     private final IotHubTransport transport;
     private final IotHubSendTask sendTask;
     private final IotHubReceiveTask receiveTask;
+    private final IotHubReconnectTask reconnectTask;
```

Comment:
Alongside the existing threads dedicated to sending and receiving messages, we'll now have a dedicate thread for responding to disconnection events